WebContent/js/utilities/sharedUtils.js

```diff
@@ -0,0 +1,20 @@
+/*
+* Each name segment (qualifier) is 1 to 8 characters, the first of which must be alphabetic (A to Z) or national (# @ $).
+* The remaining seven characters are either alphabetic, numeric (0 - 9), national, a hyphen (-). Name segments are separated by a period (.).
+* Data set names must not exceed 44 characters, including all name segments and periods.
+*/
+export default function validateName(type, nameToValidate) {
+    let regex = '';
+
+    switch (type) {
+        case 'dataset':
+            regex = /^([A-Z#@$][A-Z0-9#@$-]{0,7}(\.[A-Z#@$][A-Z0-9#@$-]{0,7})*)/g;
+            break;
+        case 'datasetMember':
+            regex = /^([A-Z#@$][A-Z0-9#@$-]{0,7})/g;
+            break;
+        default: /* renameDatasetMember */
+            regex = /^(\([A-Z#@$][A-Z0-9#@$-]{0,7}\))/g;
+    }
+    return nameToValidate.match(regex);
```

Comment:
instead of adding <= 44 checks everywhere, you can do

`return nameToValidate.match(regex) && nameToValidate.length <= 44;`

Reply:
We are returning the part of the nameToValidate that matches the regrex
So if the part that is returned is not the same as what the user has entered then we give the warning and disable the OK button
We are not returning true or false in case there is match or not

Reply:
I made the change 
    return nameToValidate.length <= 44 ? nameToValidate.match(regex) : null;